DataFormats/TrackerRecHit2D/interface/VectorHit.h

```diff
@@ -0,0 +1,160 @@
+#ifndef TrackerRecHit2D_VectorHit_h
+#define TrackerRecHit2D_VectorHit_h
+
+/** \class VectorHit
+ *
+ * 4-parameter RecHits for Phase2 Tracker (x,y, dx/dz, dy/dz)
+ *
+ * $Date: 2015/03/30 $
+ * \author Erica Brondolin
+ *
+ */
+
+#include "DataFormats/TrackerRecHit2D/interface/BaseTrackerRecHit.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+#include "DataFormats/TrackerRecHit2D/interface/OmniClusterRef.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrackingRecHit/interface/KfComponentsHolder.h"
+
+#include "TkCloner.h"
+
+class VectorHit : public BaseTrackerRecHit {
+public:
+  typedef OmniClusterRef::Phase2Cluster1DRef ClusterRef;
+
+  VectorHit() : thePosition(), theDirection(), theCovMatrix(), theDimension(0) { setType(bad); }
+
+  VectorHit(const VectorHit& vh);
+
+  VectorHit(const GeomDet& idet,
+            const LocalPoint& posInner,
+            const LocalVector& dir,
+            const AlgebraicSymMatrix& covMatrix,
+            const double& Chi2,
```

Comment:
why double?

Reply:
Can probably safely be reduced to float, done